src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -219,7 +219,15 @@
             # Can `GROUP BY coalesce(col, 1)`
             Ref("ExpressionSegment"),
         ),
-        terminator=OneOf("ORDER", "LIMIT", "HAVING", "QUALIFY", "WINDOW"),
+        terminator=OneOf(
+            "ORDER", "LIMIT", "FETCH", "OFFSET", "HAVING", "QUALIFY", "WINDOW"
```

Comment:
Had to add the `FETCH` and `OFFSET` keywords to all the appropriate terminators.

Initially considered putting it in ANSI, but the `WITH OFFSET` clause in bigquery would break so just overwrote in Snowflake dialect.